lms/djangoapps/teams/views.py

```diff
@@ -136,6 +137,19 @@ def get(self, request, course_id):
         topics = get_alphabetical_topics(course)
         organization_protection_status = user_organization_protection_status(request.user, course_key)
 
+        # We have some frontend logic that needs to know if we have any open, public, or managed teamsets,
+        # and it's easier to just figure that out here when we have them all already
+        has_open_teamset = False
+        has_public_managed_teamset = False
+        has_private_managed_teamset = False
+        for topic in topics:
+            has_open_teamset |= topic.type == TeamsetType.open
+            has_public_managed_teamset |= topic.type == TeamsetType.public_managed
+            has_private_managed_teamset |= topic.type == TeamsetType.private_managed
+            if has_open_teamset and has_public_managed_teamset and has_private_managed_teamset:
+                break
+        has_managed_teamset = has_private_managed_teamset or has_public_managed_teamset
```

Comment:
I think changing the loop to three comprehensions is a little cleaner and one line shorter? :)
has_open_teamsets = any([t for t in topics if t['type']==TeamsetType.open )
has_public_managed_teamset = any([t for t in topics if t['type']==TeamsetType.public_managed )
has_private_managed_teamset = any([t for t in topics if t['type']==TeamsetType.private_managed )

Reply:
yeah, I was just trying to avoid doing in three loops what we can do in one